### PR TITLE
Post custom classes

### PR DESCRIPTION
**Pull request by andytuba**
Add or remove class names from posts that match custom criteria a la filteReddi customFilters.

screen shot 2016-06-05 at 3 04 05 pm

screen shot 2016-06-05 at 2 37 47 pm

screen shot 2016-06-05 at 3 03 26 pm

### CODE REVIEW

File: lib/core/options/builders.js [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+						['self post', 'self'],
+						// 'comment',
+					],
+				}, '.',
+			],
+			evaluate(thing, data) {
+				switch (data.kind) {
+					case 'comment': return thing.isComment();
+					case 'link':    return thing.isSelfPost();
+					case 'self':    return this.isLinkPost();
```
Comment (erikdesjardins) [inline lib/core/options/builders.js]: `this` -> `thing`

Also these have always been backwards, apparently: `case 'link':    return thing.isSelfPost();`

File: lib/modules/stylesheet.js [hunk 2]
```diff
@@ -332,3 +342,35 @@ function shouldApply(toggle, applyTo, applyList) {
 			return true;
 	}
 }
+
+function scanEntries(ele) {
+	const things = Thing.things(ele);
+
+	things.forEach(thing => {
```
Comment (erikdesjardins) [inline lib/modules/stylesheet.js]: `for (const thing of things) {`

File: lib/modules/settingsConsole.js [hunk 3]
```diff
@@ -1346,6 +1346,7 @@ const builderFields = {
 			$addButton.append('<option value="">+ add a condition</option>');
 			let addableKeys = blockConfig.include;
 			if (addableKeys === 'all') addableKeys = Object.keys(config);
+			if (blockConfig.exclude) addableKeys = addableKeys.filter(k => blockConfig.exclude.indexOf(k) === -1);
```
Comment (erikdesjardins) [inline lib/modules/settingsConsole.js]: `!blockConfig.exclude.includes(k)`
or
`addableKeys = _.difference(addableKeys, blockConfig.exclude);`

File: lib/modules/stylesheet.js [hunk 4]
```diff
@@ -332,3 +342,35 @@ function shouldApply(toggle, applyTo, applyList) {
...
+		const [ addClassNames, removeClassNames ] = getClassNameChanges(thing);
+		if (!/^\s*$/.test(addClassNames + removeClassNames)) {
```
Comment (erikdesjardins) [inline lib/modules/stylesheet.js]: `(addClassNames + removeClassNames).trim() !== ''` might be clearer

File: lib/core/options/builders.js [hunk 5]
```diff
@@ -0,0 +1,483 @@
...
+		/*
+		possible for later:
+
+		res tag
+		res score
+		up/down/unvoted
+		sort
+		//*/
+	},
+	... options });
```
Comment (erikdesjardins) [inline lib/core/options/builders.js]: Nit: newline before `});`, no space around `...` (both will be linted soon<sup>TM</sup>)

File: lib/core/options/builders.js (hunk 1 above)
Comment (andytuba) [inline lib/core/options/builders.js]: whomp whomp.. not gonna bother migrating anyone who's worked around it.

**Comment by andytuba** [comment]
I realized `<div class="thing">` has a lot of `data-` attributes that surface most of what I wanted to do with this module, so I'm putting it on hold for a while.

**Pull request closed without merging**